dbt/adapters/sqlserver/connections.py

```diff
@@ -271,6 +272,40 @@ def get_pyodbc_attrs_before(credentials: SQLServerCredentials) -> Dict:
     return attrs_before
 
 
+def handle_datetimeoffset(dto_value) -> datetime:
```

Comment:
Would it be possible to add a unit test for this function?

Reply:
I've added a broader functional test for `dbt source freshness` for different types so this gets coverage, but haven't done a specific test for this handler logic.